src/services/catalog.service.ts

```diff
@@ -35,8 +38,46 @@ export class CatalogService {
         }
       });  // .ea category
     });  // .ea tag
+    if (_.isEmpty(catsBySubCats)) {
+      catsBySubCats[otherId] = otherId;
+    }
     return catsBySubCats;
   }
+
+  public hasCategory(item: any, category: string) {
+    return _.includes(item.catsBySubCats, category);
+  }
+
+  public hasSubCategory(item: any, subCategory: string) {
+    return _.includes(Object.keys(item.catsBySubCats), subCategory);
+  }
+
+  /**
+   * Return a new Array of only those Categories and SubCategories which
+   * exist in the passed in items.
+   * @param items
+   * @returns {Array}
+   */
+  public removeEmptyCategories(items: IServiceItem) {
```

Comment:
Hi @spadgett, I just pushed a branch which removes the rows and columns:  https://github.com/dtaylor113/origin-web-catalog/tree/removed-subcat-rows
@cshinn is going to pull this new branch and work on the css and html.